src/external/adobe_coeff.c

```diff
@@ -257,7 +257,7 @@ static void dt_dcraw_adobe_coeff(const char *name, float cam_xyz[1][12])
     { "Nikon D100", { 5902,-933,-782,-8983,16719,2354,-1402,1455,6464 } },
     { "Nikon D1H", { 7577,-2166,-926,-7454,15592,1934,-2377,2808,8606 } },
     { "Nikon D1X", { 7702,-2245,-975,-9114,17242,1875,-2679,3055,8521 } },
-    { "Nikon D1", { 16772,-4726,-2141,-7611,15713,1972,-2846,3494,9521 } }, /* multiplied by 2.218750, 1.0, 1.148438 */
+    { "Nikon D1", { 7559,-2130,-965,-7611,15713,1972,-2478,3042,8290 } },
```

Comment:
I had that, it's some caching issue.
Remove that image from your library.db, and re-import it.
Then it will stop thinking it is DT_IMAGE_4BAYER, will only look for 3 coeffs, and it should be back to normal.